internal/controller/operatorconfigmap_controller.go

```diff
@@ -469,6 +476,32 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 	return nil
 }
 
+func (c *OperatorConfigMapReconciler) getDomainLabels() []string {
+	topologyLabels := []string{
+		"kubernetes.io/hostname",
+		"topology.kubernetes.io/zone",
+		"topology.rook.io/rack",
+	}
+
+	// list node labels
+	nodes := &corev1.NodeList{}
+	err := c.List(c.ctx, nodes)
+	if err != nil {
+		c.log.Error(err, "failed to list nodes labels")
+	}
+
+	actualTopologyLabels := []string{}
+	for _, node := range nodes.Items {
+		for _, label := range topologyLabels {
+			if node.GetLabels()[label] != "" {
```

Comment:
since you are only interested in the metadata pls use PartialMetadataObject and during setting up of watches use metadataonly predicate.

Reply:
mentioned in other comments